### PR TITLE
fix: Correctly retrieve GitHub enterprise url

### LINKED ISSUE #854: Cannot get repo from enterprise github
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

I'm trying to configure an enterprise version of github and it keeps giving the error:
`ERROR: err - authentication required` and
`something went wrong in target "external-secrets" : "repository does not exist"`.

I've tried giving more permissions to the PAT, but from the documentation it's not clear what permissions should the PAT have or the format of the username.

### Expected Behavior

_No response_

### Steps To Reproduce

```
---
title: "Bump external-secrets version"
# Define git repository configuration to know where to push changes
scms:
  default:
    kind: "github"
    spec:
      # user: "{{ .github.team.user }}"
      # email: "{{ .github.team.email }}"
      owner: "{{ .github.team.owner }}"
      repository: "{{ .github.team.repository }}"
      token: "{{ requiredEnv .github.team.token }}"
      username: "{{ .github.team.username }}"
      # branch: "{{ .github.team.branch }}"
      url: "{{ .github.team.url }}"

# Defines where we get source values
sources:
  external-secrets:
    name: Get latest "external-secrets" Helm Chart Version
    kind: helmchart
    spec:
        name: external-secrets
        url: https://charts.external-secrets.io

targets:
  external-secrets:
    name: "External secrets"
    kind: "helmchart"
    scmid: "default"
    sourceid: "external-secrets"
    spec:
      file: apps/values.yaml
      key: external_secrets.source.targetRevision
      name: owner/repo
      versionincrement: minor
```

Run `updatecli diff --config updatecli/updatecli.d --values updatecli/values.yaml --debug`

And the out put says
```
SCM repository retrieved: 1
ERROR: err - authentication required
```

But running `git clone https://USERNAME:PAT_TOKEN@github.x_company.com/owner/repo` works.

### Environment

```markdown
- OS: macos 12
- updatecli: 0.33.0
```

### Anything else?

Thank you all in advance :D

### PR DESCRIPTION
**Pull request by olblak**
Signed-off-by: Olblak <me@olblak.com>

Fix #854 

## Test

To test this pull request, you can run the following commands:

```shell
# Build binary using Golang 1.19
go build -o bin/updatecli .
# Test using a manifest
./bin/updatecli diff --config updatecli/updatecli.d
```

### Tradeoff

I don't have access to a GitHub enterprise service to test. But we probably just read to quickly this doc https://github.com/shurcooL/githubv4#usage

### CODE REVIEW

**Comment by gerardsegarra** [comment]
Oh, I see you've already been working on it. I just created this PR https://github.com/updatecli/updatecli/pull/859, but I'm not an expert either in golang or updatecli code :P

**Comment by olblak** [comment]
> Oh, I see you've already been working on it. I just created this PR #859, but I'm not an expert either in golang or updatecli code :P

Your PR is better as it has tests, so I'll do some suggestion to yours :p

**Comment by olblak** [comment]
@gerardsegarra In the end I think my PR is in a more advanced state even thought we had the same thing in mind.

I improved the spec so we can also specify http protocol in the URL. If no protocol is specified then it fallbacks to https

Additional work would be needed if we want to allow the `ssh:// protocol`

File: pkg/plugins/scms/github/changelog.go [hunk 1]
```diff
@@ -68,12 +69,15 @@ func (g *Github) Changelog(version version.Version) (string, error) {
...
+		return "", err
+	}
+
 	if len(query.Repository.Release.Url) == 0 {
 		// TODO: getRepositoryURL()
-		return fmt.Sprintf("No Github Release found for %s on https://github.com/%s/%s",
-			versionName,
-			g.Spec.Owner,
-			g.Spec.Repository), nil
+		return fmt.Sprintf("no Github Release found for %s on %q", versionName, URL), nil
```
Comment (lemeurherve) [inline pkg/plugins/scms/github/changelog.go:80]:
```suggestion
		return fmt.Sprintf("no GitHub Release found for %s on %q", versionName, URL), nil
```

File: pkg/plugins/scms/github/changelog_test.go [hunk 2]
```diff
@@ -70,7 +70,7 @@ func TestChangelog(t *testing.T) {
 					Release: queriedRelease{},
 				},
 			},
-			wantChangelog: "No Github Release found for v0.17.0 on https://github.com/updatecli/updatecli",
+			wantChangelog: "no Github Release found for v0.17.0 on \"https://github.com/updatecli/updatecli\"",
```
Comment (lemeurherve) [inline pkg/plugins/scms/github/changelog_test.go:73]:
```suggestion
			wantChangelog: "no GitHub Release found for v0.17.0 on \"https://github.com/updatecli/updatecli\"",
```

**Review by lemeurherve — APPROVED** [review]
GitHub correct case (nit)
LGTM otherwise

**Review by lemeurherve — APPROVED** [review]

**Comment by olblak** [comment]
Thanks @lemeurherve for the review and thanks @gerardsegarra for spotting this issues

**Comment by gerardsegarra** [comment]
Thank you so much for fixing it! 😄

**Pull request merged**